### PR TITLE
[execution] Tests should use common utilities for managing accounts

### PR DESCRIPTION
**Pull request by ikabiljo**
### Description
https://github.com/aptos-labs/aptos-core/issues/353

### Test Plan
cargo x test -p executor -- test_execution_with_storage
cargo xtest -p executor-benchmark --lib   

cargo xfmt
cargo xlint 
cargo xclippy --workspace --all-targets

### CODE REVIEW

File: execution/executor-benchmark/src/transaction_generator.rs [hunk 1]
```diff
@@ -310,15 +292,15 @@ impl TransactionGenerator {
                         .index(0);
                 let seed_account = &self.seed_accounts_cache[seed_index];
                 let txn = create_transaction(
-                    &seed_account.private_key,
-                    seed_account.public_key.clone(),
+                    seed_account.private_key(),
```
Comment (zekun000) [inline execution/executor-benchmark/src/transaction_generator.rs:295]: I think we can replace this `create_transaction` with `sign_with_transaction_builder` from the local account

File: execution/executor-test-helpers/Cargo.toml [hunk 2]
```diff
@@ -29,3 +29,4 @@ move-deps = { path = "../../aptos-move/move-deps", features = ["address32"] }
 storage-interface = { path = "../../storage/storage-interface" }
 storage-service = { path = "../../storage/storage-service" }
 vm-genesis = { path = "../../aptos-move/vm-genesis" }
+aptos-sdk = { path = "../../sdk" }
```
Comment (zekun000) [inline execution/executor-test-helpers/Cargo.toml:32]: there's linter error on this not being sorted

File: execution/executor-test-helpers/src/integration_test_impl.rs [hunk 3]
```diff
@@ -48,117 +54,70 @@ pub fn test_execution_with_storage_impl() -> Arc<AptosDB> {
...
     // Create 14 txns transferring 10k from account1 to account3 each.
-    for i in 2..=15 {
-        block2.push(get_test_signed_transaction(
-            account1,
-            /* sequence_number = */ i,
-            privkey1.clone(),
-            pubkey1.clone(),
-            Some(aptos_stdlib::encode_test_coin_transfer(account3, 10_000)),
-        ));
+    for _i in 2..=15 {
```
Comment (zekun000) [inline execution/executor-test-helpers/src/integration_test_impl.rs:117]: we can do `for _ in 2..=15`

**Review by zekun000 — COMMENTED** [review]
looks awesome!
nit: we should update the PR title similar to the commit message

**Review by zekun000 — APPROVED** [review]

**Comment by ikabiljo** [comment]
@msmouse updated transaction_generator.rs as well in https://github.com/aptos-labs/aptos-core/pull/1365 to use new API and support more transactions, so I'll remove it from this diff

**Pull request closed without merging**